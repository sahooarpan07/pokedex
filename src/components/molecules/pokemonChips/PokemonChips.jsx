import {View, Text} from 'react-native';
import React from 'react';
import PokemonChip from '../../atoms/pokemonChip/PokemonChip';
import {COLORS} from '../../../constants/constants';
import {getCapitalisedWord} from '../../../util/utils';

const PokemonChips = ({pokemon, showWeakness}) => {
  return (
    <View style={{flexDirection: 'row', gap: 8}}>
      {pokemon?.types.map((type, index) => {
        console.log(pokemon.weakness, 'selectedPokemon.weakness in chips');
        const value =
          showWeakness && pokemon.weakness
            ? pokemon.weakness[index]
            : type?.type?.name;
        console.log('value', value);
        return (
          <PokemonChip
            backgroundColor={COLORS[type?.type?.name]}
            value={value ? getCapitalisedWord(value) : type.type.name}
            key={index}
          />
        );
      })}
    </View>
  );
};

export default PokemonChips;
