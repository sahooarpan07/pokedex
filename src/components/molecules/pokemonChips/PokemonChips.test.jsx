/**
 * @format
 */

import 'react-native';
import React from 'react';
import PokemonChips from './PokemonChips';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders Pokemon Card correctly', () => {
  const types = [
    {
      slot: 1,
      type: {
        name: 'grass',
        url: 'https://pokeapi.co/api/v2/type/12/',
      },
    },
    {
      slot: 2,
      type: {
        name: 'poison',
        url: 'https://pokeapi.co/api/v2/type/4/',
      },
    },
  ];
  const pokemon = {
    id: 1,
    name: 'Bulbasuar',
    types,
  };

  const tree = renderer.create(<PokemonChips pokemon={pokemon} />).toJSON();

  expect(tree).toMatchSnapshot();
});
