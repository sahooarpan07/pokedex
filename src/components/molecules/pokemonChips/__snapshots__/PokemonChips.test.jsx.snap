// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pokemon Card correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "gap": 8,
    }
  }
>
  <View
    style={
      [
        {
          "borderColor": "#180345",
          "borderRadius": 6,
          "borderWidth": 2,
          "padding": 4,
        },
        {
          "backgroundColor": "#C0D4C8",
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#180345",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      Grass
    </Text>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#180345",
          "borderRadius": 6,
          "borderWidth": 2,
          "padding": 4,
        },
        {
          "backgroundColor": "#CFB7ED",
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#180345",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      Poison
    </Text>
  </View>
</View>
`;
